jobs/src/main/scala/dcos/metronome/model/JobRunSpec.scala

```diff
@@ -71,6 +73,8 @@ object JobRunSpec {
       def noSecretVolumesExists: Boolean = jobRunSpec.volumes.forall(v => !isSecretVolume(v))
       check(noSecretVolumesExists || jobRunSpec.ucr.isDefined, JobRunSpecMessages.fileBasedSecretsAreUcrOnly)
 
+      check(jobRunSpec.gpus == 0 || jobRunSpec.ucr.isDefined, JobRunSpecMessages.gpusAreUcrOnly)
```

Comment:
this check seems off...    we shouldn't short circuit with `||`... if gpus are > 0.... ucr.isDefined isn't evaluated because it wouldn't matter whats value is for the value to be true.    It also allows for gpus > 0 and ucr to not be defined.   We want an ADD here.

Reply:
haha... ok.. I'm good with it.   there is a lot of negations in this code... it is challenging to read and mentally validate...